src/Common/Extensions/CmdletExtensions.cs

```diff
@@ -232,6 +232,11 @@ public static string TryResolvePath(this PSCmdlet psCmdlet, string path)
             }
         }
 
+        public static string TryResolvePathWithException(this PSCmdlet psCmdlet, string path)
```

Comment:
Generally, the pattern is that a method with the name 'Try*' does not throw exceptions.  What is this method actually doing?